packages/api/tests/lambdas/test-cleanExecutions.js

```diff
@@ -1,58 +1,470 @@
 
-const { cleanExecutionPayloads } = require('../../lambdas/cleanExecutions');
-
-const removeRecordsStub = sinon.spy();
+  t.context.knex = knex;
+  t.context.knexAdmin = knexAdmin;
+  const { esIndex, esClient, searchClient } = await createTestIndex();
+  t.context.esIndex = esIndex;
+  t.context.esClient = esClient;
+  t.context.searchClient = searchClient;
+  const records = [];
+  for (let i = 0; i < 10; i += 1) {
+    records.push(fakeExecutionRecordFactory({
+      updated_at: moment().subtract(i, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+      original_payload: '{"b": "c"}',
+      status: 'completed',
+    }));
+    records.push(fakeExecutionRecordFactory({
+      updated_at: moment().subtract(i, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+      original_payload: '{"b": "c"}',
+      status: 'failed',
+    }));
+  }
+  const model = new ExecutionPgModel();
 
-class executionModel {
-  removeOldPayloadRecords(...value) {
-    return removeRecordsStub(...value);
+  const pgRecords = await model.insert(t.context.knex, records, '*');
+  for (const record of pgRecords) {
+    await t.context.esClient.client.index({
+      body: await translatePostgresExecutionToApiExecution(record),
+      id: record.cumulus_id,
+      index: t.context.esIndex,
+      type: 'execution',
+      refresh: true,
+    });
   }
-}
+  t.context.execution_cumulus_ids = pgRecords.map((record) => record.cumulus_id);
+});
 
-test.beforeEach(() => {
-  process.env.completeExecutionPayloadTimeoutDisable = false;
-  process.env.nonCompleteExecutionPayloadTimeoutDisable = false;
+test.afterEach.always(async (t) => {
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    tesetDbName: t.context.testDbName,
+  });
+  await cleanupTestIndex(t.context);
 });
 
-test.serial('Function is called with correct timeout values', async (t) => {
-  process.env.completeExecutionPayloadTimeout = 100;
-  process.env.nonCompleteExecutionPayloadTimeout = 50;
-  await cleanExecutionPayloads(executionModel);
-  t.truthy(removeRecordsStub.calledWith(100, 50, false, false));
+const pgPayloadsEmpty = (entry) => !entry.final_payload && !entry.orginal_payload;
+
+const esPayloadsEmpty = (entry) => !entry.finalPayload && !entry.orginalPayload;
+
+test.serial('handler() handles complete expiration', async (t) => {
+  const env = clone(process.env);
+  process.env = localStackConnectionEnv;
+  process.env.PG_DATABASE = t.context.testDbName;
+  process.env.ES_INDEX = t.context.esIndex;
+  process.env.LOCAL_ES_HOST = 'localhost';
+  let expirationDays = 4;
+  let {
+    completeExpiration: expirationDate,
+  } = getExpirationDates(
+    expirationDays,
+    0,
+    true,
+    false
+  );
+  process.env.completeExecutionPayloadTimeoutDisable = 'false';
+  process.env.nonCompleteExecutionPayloadTimeoutDisable = 'true';
+  process.env.completeExecutionPayloadTimeout = expirationDays;
+
+  await handler();
+  await sleep(5000);
```

Comment:
Hmm this seems like it could be fragile. Is there a better way to ensure the handler has completed?